### PR TITLE
FreeDrive event location fix

### LINKED ISSUE #3786: NullPointerException when formatting telemetry date
Might be related to https://github.com/mapbox/mapbox-navigation-android/issues/3765. Reproduced on `v1.3.0-alpha.1`.

```
11-17 15:40:21.464 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: MapboxMetricsReporter.init from MapboxNavigation main
11-17 15:40:21.467 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: onOffRouteStateChanged false
11-17 15:40:21.467 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: session state is IDLE
11-17 15:40:21.467 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: sessionStop
11-17 15:40:21.467 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: resetOriginalRoute
11-17 15:40:21.467 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: resetRouteProgress
11-17 15:40:21.467 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: Valid initialization
11-17 15:40:21.488 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: session state is FREE_DRIVE
11-17 15:40:21.488 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: sessionStop
11-17 15:40:21.488 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: resetOriginalRoute
11-17 15:40:21.488 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: resetRouteProgress
11-17 15:40:21.488 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: trackFreeDrive START
11-17 15:40:21.488 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: wait for location
11-17 15:40:21.497 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: Lifecycle monitor created
11-17 15:40:21.497 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: location not ready
11-17 15:40:21.501 4540-4540/com.mapbox.navigation.instrumentation_tests W/LocationUpdateReceiver: LocationEngineResult == null
11-17 15:40:21.516 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: session state is ACTIVE_GUIDANCE
11-17 15:40:21.516 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: flush buffer. Pending events = 0
11-17 15:40:21.516 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: trackFreeDrive STOP
11-17 15:40:21.516 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: createFreeDriveEvent STOP
11-17 15:40:21.518 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: populateNavigationFreeDriveEvent
11-17 15:40:21.519 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: MapboxNavigation onDestroy
11-17 15:40:21.520 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: session state is IDLE
11-17 15:40:21.520 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: sessionStop
11-17 15:40:21.520 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: resetOriginalRoute
11-17 15:40:21.520 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: resetRouteProgress
11-17 15:40:21.520 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: trackFreeDrive STOP
11-17 15:40:21.520 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: createFreeDriveEvent STOP
11-17 15:40:21.522 4540-4562/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: populateNavigationFreeDriveEvent
11-17 15:40:21.584 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: location ready
11-17 15:40:21.584 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: createFreeDriveEvent START
11-17 15:40:21.584 1565-1583/system_process W/ActivityManager: Unable to start service Intent { act=com.google.android.gms.nearby.discovery:ACTION_PERMISSION_CHANGE cmp=com.google.android.gms/.nearby.discovery.service.DiscoveryService } U=0: not found
11-17 15:40:21.586 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: populateNavigationFreeDriveEvent
11-17 15:40:21.586 4540-4540/com.mapbox.navigation.instrumentation_tests D/MAPBOX_TELEMETRY: class com.mapbox.navigation.core.telemetry.events.NavigationFreeDriveEvent event sent
11-17 15:40:21.587 4540-4540/com.mapbox.navigation.instrumentation_tests D/LifecycleMonitor: Lifecycle status change: com.mapbox.navigation.instrumentation_tests.activity.EmptyTestActivity@3ca8b5b3 in: PAUSED
11-17 15:40:21.604 4540-4562/com.mapbox.navigation.instrumentation_tests E/TestRunner: java.lang.NullPointerException: Attempt to invoke virtual method 'long java.util.Date.getTime()' on a null object reference
        at java.util.Calendar.setTime(Calendar.java:1195)
        at java.text.SimpleDateFormat.formatImpl(SimpleDateFormat.java:518)
        at java.text.SimpleDateFormat.format(SimpleDateFormat.java:820)
        at java.text.DateFormat.format(DateFormat.java:307)
        at com.mapbox.android.telemetry.TelemetryUtils.generateCreateDateFormatted(TelemetryUtils.java:190)
        at com.mapbox.navigation.core.telemetry.MapboxNavigationTelemetry.populate(MapboxNavigationTelemetry.kt:533)
        at com.mapbox.navigation.core.telemetry.MapboxNavigationTelemetry.access$populate(MapboxNavigationTelemetry.kt:100)
        at com.mapbox.navigation.core.telemetry.MapboxNavigationTelemetry$trackFreeDrive$1.invoke(MapboxNavigationTelemetry.kt:296)
        at com.mapbox.navigation.core.telemetry.MapboxNavigationTelemetry.trackFreeDrive(MapboxNavigationTelemetry.kt:322)
        at com.mapbox.navigation.core.telemetry.MapboxNavigationTelemetry.handleStateChanged(MapboxNavigationTelemetry.kt:277)
        at com.mapbox.navigation.core.telemetry.MapboxNavigationTelemetry.onNavigationSessionStateChanged(MapboxNavigationTelemetry.kt:260)
        at com.mapbox.navigation.core.NavigationSession.setState(NavigationSession.kt:22)
        at com.mapbox.navigation.core.NavigationSession.updateState(NavigationSession.kt:42)
        at com.mapbox.navigation.core.NavigationSession.setDriving(NavigationSession.kt:37)
        at com.mapbox.navigation.core.NavigationSession.onSessionStateChanged(NavigationSession.kt:71)
        at com.mapbox.navigation.core.trip.session.MapboxTripSession.setState(MapboxTripSession.kt:123)
        at com.mapbox.navigation.core.trip.session.MapboxTripSession.stop(MapboxTripSession.kt:201)
        at com.mapbox.navigation.core.MapboxNavigation.onDestroy(MapboxNavigation.kt:351)
        at com.mapbox.navigation.core.MapboxNavigationProvider.destroy(MapboxNavigationProvider.kt:49)
        at com.mapbox.navigation.instrumentation_tests.utils.MapboxNavigationRule.finished(MapboxNavigationRule.kt:18)
```

/cc @korshaknn @Guardiola31337

### PR DESCRIPTION
**Pull request by korshaknn**
### Description

closes #3786
We don't wait for a location to send `FreeDriveEvent`
If `location` is not available, `null` will be used

### Changelog

```
<changelog>Pass `null` to FreeDriveEvent if `location` is not available</changelog>
```

### CODE REVIEW

**Review by LukasPaczos — APPROVED** [review]
LGTM, could you also add a test that reproduces the previous crash?
1. Start free drive
2. Go straight to active guidance

**Comment by Guardiola31337** [comment]
From my tests `START` events never have `location` (is always `null`) :disappointed:

We should check when the `locationsCollector` observer is registered because maybe trip session doesn't provide a location until the session is started but when it is started we immediately send event. The issue may be fixed if we do it after registering 🤔 

cc @korshaknn

**Comment by korshaknn** [comment]
@LukasPaczos added more tests

File: libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetryTest.kt [hunk 1]
```diff
@@ -600,6 +638,21 @@ class MapboxNavigationTelemetryTest {
         assertEquals(STOP.type, stopFreeDriveEvent.eventType)
     }
 
+    @Test
+    fun freeDrive_start_and_stop_sent_when_state_changes_from_free_drive_to_active_guidance() {
+        baseMock()
```
Comment (LukasPaczos) [inline libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetryTest.kt:643]:
```suggestion
    fun `freeDrive start+stop sent when state changes from free drive to active guidance`() {
        baseMock()
        every { locationsCollector.lastLocation } returns null
```

The location has to be null to reproduce the issue.

Comment (korshaknn) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetryTest.kt:643]: check `freeDrive_sent_with_null_location_when_location_not_available_from_free_drive_to_active`

Comment (LukasPaczos) [reply libnavigation-core/src/test/java/com/mapbox/navigation/core/telemetry/MapboxNavigationTelemetryTest.kt:643]: :+1:

**Comment by Guardiola31337** [comment]
```
2020-11-18 11:29:20.995 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: Valid initialization
2020-11-18 11:29:43.459 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: session state is FREE_DRIVE
2020-11-18 11:29:43.459 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: sessionStop
2020-11-18 11:29:43.459 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: resetOriginalRoute
2020-11-18 11:29:43.459 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: resetRouteProgress
2020-11-18 11:29:43.459 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: trackFreeDrive START
2020-11-18 11:29:43.460 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: createFreeDriveEvent START
2020-11-18 11:29:43.476 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: populateFreeDriveEvent
2020-11-18 11:29:43.477 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: class com.mapbox.navigation.core.telemetry.events.NavigationFreeDriveEvent event sent
2020-11-18 11:29:43.504 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: Lifecycle monitor created
2020-11-18 11:29:43.509 25573-25573/com.mapbox.navigation.examples D/MAPBOX_TELEMETRY: onRawLocationChanged
```

`onRawLocation` is received after `onNavigationSessionStateChanged` when the event has been already sent.

**Comment by Guardiola31337** [comment]
```kotlin
override fun start() {
        if (state == TripSessionState.STARTED) {
            return
        }
        tripService.startService()
        startLocationUpdates()
        state = TripSessionState.STARTED
    }
```
`startLocationUpdates` is async and `state` executes the `onSessionStateChanged` right away before `onRawLocationChanged`.

We should investigate a different way to wait for the first location https://github.com/mapbox/mapbox-navigation-android/pull/3751#discussion_r521379048 and also discuss what to do if a session is started and no locations are received. Refs. https://github.com/mapbox/mapbox-navigation-android/issues/2827

cc @korshaknn @kmadsen @LukasPaczos

**Comment by Guardiola31337** [comment]
Let's merge for now @korshaknn

**Comment by Guardiola31337** [comment]
Cut a follow up ticket https://github.com/mapbox/mapbox-navigation-android/issues/3797 so we don't forget.

**Pull request merged**